server/src/main/java/org/elasticsearch/index/mapper/DynamicRuntimeFieldsBuilder.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.common.time.DateFormatter;
+import org.elasticsearch.plugins.MapperPlugin;
+
+/**
+ * Defines how runtime fields are created dynamically. Used when objects are mapped with dynamic:runtime.
+ * @see MapperPlugin#getDynamicRuntimeFieldsBuilder()
+ */
+public interface DynamicRuntimeFieldsBuilder {
```

Comment:
Could we use `DynamicFieldsBuilder` instead? And let the runtime plugin delegate for binary fields? Should we throw an exception for binary fields instead? Or not create them at all? It fields weird that `runtime:dynamic` could make non-dynamic fields.

Reply:
This one can only even return `RuntimeFieldType`, hence runtime in its name seems a good fit? the binary field situation is indeed a bit weird. The point of having this different interface for plugins is that the return type can only be RuntimeFieldType. The other internal abstraction returns DynamicField which is really some internal shenanigans to wrap either a mapper or a runtime field type, something that I would not want to expose to plugins.

Reply:
Yeah, that makes sense. I you are calling it `dynamic: runtime`. It'd be weird to return something else!